pkg/deploy/kubernetes_image_puller.go

```diff
@@ -171,27 +179,36 @@ func ReconcileImagePuller(ctx *DeployContext) (reconcile.Result, error) {
 
 	} else {
 		if foundOperatorsAPI && foundPackagesAPI {
-			removed, err := UninstallImagePullerOperator(ctx)
-			if err != nil {
-				logrus.Errorf("Error uninstalling Image Puller: %v", err)
-				return reconcile.Result{}, err
+			reconcileResult, err := DeelteImagePullerOperatorAndFinalizer(ctx)
+			if reconcileResult != nil {
+				return *reconcileResult, err
 			}
+		}
+	}
+	return reconcile.Result{}, nil
+}
 
-			if removed {
-				return reconcile.Result{RequeueAfter: time.Second}, nil
-			}
+func DeelteImagePullerOperatorAndFinalizer(ctx *DeployContext) (*reconcile.Result, error) {
+	removed, err := UninstallImagePullerOperator(ctx)
+	if err != nil {
+		logrus.Errorf("Error uninstalling Image Puller: %v", err)
+		return &reconcile.Result{}, err
+	}
 
-			if HasImagePullerFinalizer(ctx.CheCluster) {
-				err = DeleteImagePullerFinalizer(ctx)
-				if err != nil {
-					logrus.Errorf("Error deleting finalizer: %v", err)
-					return reconcile.Result{}, err
-				}
-				return reconcile.Result{RequeueAfter: time.Second}, nil
-			}
+	if removed {
+		return &reconcile.Result{RequeueAfter: time.Second}, nil
+	}
```

Comment:
Do we really need this?
That's not obvious.

Reply:
After some testing, it can be removed